easy_my_coop/models/subscription_request.py

```diff
@@ -75,10 +75,21 @@ def _check_share_available_to_user(self):
                     % request.share_product_id.name
                 )
 
+    def _send_confirmation_email(self):
+        if self.company_id.send_confirmation_email:
+            mail_template_notif = self.get_mail_template_notif(
+                is_company=self.partner_id.is_company
+            )
+            mail_template_notif.sudo().send_mail(self.id)
```

Comment:
it’s good to extract this to a separate method. 👍 i would go even further by defining another method:
```python
def _create_subscription_request_with_confirmation(self, vals):
    subscription_request = super(SubscriptionRequest, self).create(vals)
    subscription_request._send_confirmation_email()
    return subscription_request
```
and call it from the 2 places.

Reply:
I see the point of extracting one step beyond but I don't like this calling the refactoring proposed because the super call is hidden in a sub_method. More over, I think the bad design here is that `create_comp_sub_req` calls `super().create` and then calls `_send_confirmation_mail`.

I wrote that down in a todo comment above the create function:

```
    # todo code this in a more pythonic way
    #  - do not bypass self.create in create_comp_sub_req
    #  - use "not in" where applicable
    #  - sanitize declaration and assignation of cooperator
    #  beware if usage in investor_wallet_platform
```

and created a task to include this in komunigi refactoring: https://gestion.coopiteasy.be/web#id=7877&view_type=form&model=project.task&action=475&active_id=221